.github/dependabot.yml

```diff
@@ -1,6 +1,9 @@
 version: 2
 updates:
-  - package-ecosystem: "github-actions"
+  - &github-actions
```

Comment:
ist das `&github-actions` und  `<<: *github-actions`  nur für die Beschreibung, dass dies ein Block ist, oder hat das noch eine andere Funktion?

Reply:
Das `&github-actions` ist in yaml ein sogenannter "anchor", das `*github-actions` ist ein "alias", und `<<:` ist die "map merging" Syntax. Mit einem alias kann man einen anchor referenzieren, um den Inhalt des anchors wiederzuverwenden. Das map merging fügt den Inhalt des dort angegebenen Werts auf der selben Ebene ein.

Konkret bedeutet das hier: `updates:` ist eine Liste mit zwei Elementen. Das erste Element ist ganz gewöhnlich angegeben, hat nur zusätzlich den Namen "github-actions". Das zweite Listenelement nutzt `<<:` um den Inhalt von "github-actions" an dieser Stelle wiederzuverwenden. Anschließend wird nur der target-branch überschrieben.

Das ist soweit ich weiß die einzige Möglichkeit, an der Stelle Duplizierung zu vermeiden.

---

The `&github-actions` is a yaml "anchor", `*github-actions` is an "alias", and `<<:` is the "map merging" syntax. An anchor is used to give an element a name which can then be reused in other places with an alias, while the map merging merges the values that are supplied into the location where it is specified.

In this case, `updates:` is a list with two elements. The first one is specified as usual, but also has a name "github-actions". The second element applies `<<:` to reuse everything from the first (via the alias) and then additionally sets target-branch.

This is AFAIK the only way to avoid duplication in yaml.